src/memory.cpp

```diff
@@ -30,44 +30,90 @@
 
 inline namespace CUPLA_ACCELERATOR_NAMESPACE
 {
+    /** allocate memory
+     *
+     * @param ptrptr Address to store the address of the allocated chunk.
+     *               Will be set to nullptr if allocation is failing.
+     * @param size Aize to allocate (in byte).
+     *
+     * @return cuplaSuccess if allocation succeed, else cuplaErrorMemoryAllocation.
+     */
```

Comment:
This and other doxygen comments added in this PR should go to declarations (in `memory.hpp`) and not here